extensions/authentication-passport/src/__tests__/acceptance/authentication-with-passport-strategy-adapter.acceptance.ts

```diff
@@ -214,18 +368,124 @@ describe('Basic Authentication', () => {
   }
 });
 
+//
+// A simple User model
+//
+interface MyUser {
+  id: string;
+  username: string;
+  firstName?: string;
+  lastName?: string;
+}
+
 class UserRepository {
   constructor(
-    readonly list: {[key: string]: {profile: UserProfile; password: string}},
+    readonly list: {[key: string]: {profile: MyUser; password: string}},
   ) {}
+
+  /*
+   *   Used in 'testing' scenario where authentication strategy does not come from a provider.
+   */
   find(username: string, password: string, cb: Function): void {
     const userList = this.list;
     function search(key: string) {
-      return userList[key].profile[securityId] === username;
+      return userList[key].profile.username === username;
     }
     const found = Object.keys(userList).find(search);
     if (!found) return cb(null, false);
     if (userList[found].password !== password) return cb(null, false);
     cb(null, userList[found].profile);
   }
+
+  /*
+   *   Used in 'testing' scenario where authentication strategy comes from a provider.
+   *   (Specifically used by verify function provider : VerifyFunctionProvider)
+   */
+  find2(username: string, password: string): MyUser | undefined {
```

Comment:
A more descriptive name? Like `findWithoutCb`? Or use function overload to distinguish the two signatures.

Reply:
I did provide a comment to distinguish them, but I will rename the 2nd one to make it clearer.